### PR TITLE
Refactor.architecture

### PR DESCRIPTION
**Pull request by herman-rogers**
Refactored the logging section to test out how to split the codebase up. A lot of this is WIP but since the overall refactor is going to be massive - I'll continue this work in other branches to make sure the code review doesn't balloon.

### CODE REVIEW

File: pkg/logger/filelogger.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+	logfile := logSave.getLogFile(logSave.LogFileName)
+	defer logfile.Close()
+	log.SetOutput(logfile)
+	log.Print(data)
+}
+
+func (logSave FileLogSave) getLogFile(file string) *os.File {
+	directory := logSave.defaultDirectory()
+	filename := logSave.defaultFile()
+	fullPath := directory + filename
```
Comment (chvck) [inline pkg/logger/filelogger.go:30]: Could be an idea to cache the path? Just lighten the I/O load ever so slightly, may not matter unless we start logging loads...

File: pkg/logger/loggerinterface.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+// SystemLogTag is the identifying name for
+// system messages in the log files
+const SystemLogTag = "SYSTEM_LOG_INFO"
+
+// SystemErrorTag is the identifying name for
+// system errors in the log files
+const SystemErrorTag = "SYSTEM_ERROR"
+
+const logTimeStamp = "Mon Jan _2 15:04:05 UTC 2006"
```
Comment (chvck) [inline pkg/logger/loggerinterface.go:29]: This still in for testing reasons?

File: pkg/logger/loggerinterface.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+	time := time.Now().Local()
+	timeString := time.Format(logTimeStamp)
+	systemInfo := SystemLogTag + " " + timeString + ": " + data
+	saveLogData(log.LogSave, systemInfo)
+	return systemInfo
+}
+
+// Error system log returns error data in the
+// format SYSTEM_ERROR, time, message
+func (log SystemLogger) Error(data string) string {
```
Comment (chvck) [inline pkg/logger/loggerinterface.go:51]: Might be worth adding Debug too?

File: pkg/logger/filelogger.go (hunk 1 above)
Comment (herman-rogers) [reply pkg/logger/filelogger.go:30]: Yea we can definitely do that, didn't want to focus to much on the infrastructure in this first pass. Could definitely be another PR with a more optimised file logger for sure.

File: pkg/logger/loggerinterface.go (hunk 2 above)
Comment (herman-rogers) [reply pkg/logger/loggerinterface.go:29]: That's how go parses timestamps apparently, the returned result would be Mon Jan 30 13:17:00 UTC 2017

File: pkg/logger/loggerinterface.go (hunk 3 above)
Comment (herman-rogers) [reply pkg/logger/loggerinterface.go:51]: yea that's the nice thing about this new architecture stuff - we can have loads of different types of loggers. I think for the app we need a UserLogger too (for information to send back to users about their cloning data). Just wanted to get the architecture in before expanding on the use cases.

**Pull request merged**